src/lsst/cmservice/client/wrappers.py

```diff
@@ -247,23 +247,64 @@ def get_object_by_fullname_function(
     Returns
     -------
     the_function: Callable
-        Function that returns a single row from a table by name
+        Function that returns a single row from a table by fullname
     """
 
-    def get_obj_by_fullname(
+    def get_row_by_fullname(
         obj: CMClient,
         fullname: str,
     ) -> response_model_class:
         params = models.FullnameQuery(
             fullname=fullname,
         )
         results = obj.client.get(f"{query}", params=params.model_dump()).json()
+        if "detail" in results:
+            return None
+        try:
+            return parse_obj_as(response_model_class, results)
+        except ValidationError as msg:
+            raise ValueError(f"Bad response: {results}") from msg
+
+    return get_row_by_fullname
+
+
+def get_row_by_name_function(
+    response_model_class: TypeAlias = BaseModel,
+    query: str = "",
+) -> Callable:
+    """Return a function that gets a single row from a table (by name)
+    and attaches that function to a client.
+
+    Parameters
+    ----------
+    response_model_class: TypeAlias = BaseModel,
+        Pydantic class used to serialize the return value
+
+    query: str
+        http query
+
+    Returns
+    -------
+    the_function: Callable
+        Function that returns a single row from a table by name
+    """
+
+    def get_row_by_name(
+        obj: CMClient,
+        name: str,
+    ) -> response_model_class | None:
+        params = models.NameQuery(
+            name=name,
+        )
+        results = obj.client.get(f"{query}", params=params.model_dump()).json()
+        if "detail" in results:
```

Comment:
Same puzzle as above.  NO DETAILS FOR YOU!

Reply:
Is there a better way to see if `obj.client.get(f"{query}", params=params.model_dump())` is failing?   It doesn't throw or return None like a normal person..